feat(app-store): add facetime app to app-store

Mythie opened the pull request:
## What does this PR do?

Add the FaceTime app to the app-store allowing for bookings locations using FaceTime web links and FaceTime app link schemes.

Fixes #6965

[Loom Video](https://www.loom.com/share/8edb6e61a7f14e41bbe2df3d5f9c97b1)

**Environment**: Staging(main branch) / Production

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Add the FaceTime app
- [ ] Use FaceTime as a booking location
- [ ] Verify you are able to initial a FaceTime call using the booking link

## Checklist

Mythie committed:
feat(app-store): add facetime app to app-store

Add the facetime app to the app-store allowing for facetime web links and facetime app link schemes.

fixes #6965

PeerRich commented:
thank you so much! feel free to change author to youe github username and not Cal.com Inc

PeerRich commented on packages/app-store/facetime/config.json:
```diff
@@ -0,0 +1,26 @@
+  "url": "https://cal.com/apps/facetime",
+  "variant": "conferencing",
+  "categories": ["video"],
+  "publisher": "Cal.com, Inc",
```
```suggestion
  "publisher": "Mythe",
```

PeerRich committed:
Update packages/app-store/facetime/config.json

hariombalhara commented on packages/app-store/facetime/components/InstallAppButton.tsx:
```diff
@@ -0,0 +1,18 @@
+import type { InstallAppButtonProps } from "@calcom/app-store/types";
```
We don't need to add this Component separately now. It's automatically handled. Can you try deleting this file and see if it works still.

hariombalhara replied:
Removed this file and _metadata.ts. Those files are not required now. New apps created using cli don't have them.
There is an issue in which we need to cleanup old apps https://github.com/calcom/cal.com/issues/6714

hariombalhara commented on packages/app-store/facetime/config.json:
```diff
@@ -0,0 +1,26 @@
+      "linkType": "static",
+      "type": "integrations:facetime_video",
+      "label": "Facetime",
+      "organizerInputPlaceholder": "https://facetime.apple.com/join... or facetime:<contact>",
```
facetime:<contact> would only work if the facetime app is installed. Otherwise this link would just not work.

So, I think we shouldn't suggest that. If facetime works through browser that's the right thing to do. 
I hope facetime website would automatically open facetime app if it's installed.

Mythie replied:
It does open the app when visiting the website, however that still requires that a user generate a FaceTime link. If we’re happy for that to be required I can make the update in a few hours once it’s morning again.

hariombalhara replied:
I think that would be a better approach because the organizer can generate the link once. If we allow **facetime:** scheme, bookers won't be able to join meeting by clicking on the link they receive.

Mythie replied:
Latest commit updates the config to follow the new decision on the FaceTime links, the placeholder is also updated correspondingly:

<img width="585" alt="image" src="https://user-images.githubusercontent.com/13398220/219810417-830a1229-0786-4d83-940f-a19ab33684cd.png">

hariombalhara committed:
Remove unneeded files and make app follow the new app rules

Mythie committed:
fix(app-store): update facetime app config

Update the regex and placeholder for the facetime app to reflect the decisions to only accept https://facetime.apple.com links. This was decided as it means that all participants can join a meeting even in the event that they don't have an Apple device.
